include/sundials/sundials.h

```diff
@@ -0,0 +1,32 @@
+/* -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2022, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------
+ * Header file that includes the SUNDIALS core.
```

Comment:
I like to think of `package.h` as an all-in-one header file that includes everything, whereas `package_core.h` as a header that only includes the essential components, leaving other features to be included as needed.

It looks to me like this file might include all non-C++ (e.g., `sundials_hip_policies.hpp`) and non-optional (e.g. `sundials_xbraid.h`) header files in the `sundials` folder, leading me to prefer `sundials.h`.  However, if I'm missing something then perhaps `sundials_core.h` would be preferred.

Reply:
I also think of `package.h` as being an all-in-one header but where the user would not need to include anything other than `package.h` to use the package. This header only contains common includes for shared sundials components and I think a user might expect `sundials.h` to provide access to all the capabilities of sundials.

Reply:
Indeed.  Since this doesn't `#include` individual solvers that are always built with SUNDIALS (e.g., `sunlinsol_dense.h` and `sunnonlinsol_fixedpoint.h`), then I agree that `sundials_core.h` (and similarly `sundials_core.hpp`) make the most sense.

Reply:
I agree with `sundials_core.h`.